File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.go
func makeEqual(words []string) bool {
    charCounts := make(map[string]int)

    for i := range len(words) {
        for j := range len(words[i]) {
            charCounts[string(words[i][j])]++
        }
    }

    for _, v := range charCounts {
        if v % len(words) != 0 {
            return false
        }
    }
    return true
}

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        map<char, int> charCounts;
        for (auto const &word : words) {
            for (const char c : word) {
                ++charCounts[c];
            }
        }
        for (auto const &[key, val] : charCounts) {
            if (val % words.size() != 0) {
                return false;
            }
        }
        return true;
    }
};